examples/run_CPMG.py

```diff
@@ -4,149 +4,184 @@
 
 This script will perform a standard CPMG experiment. 
 In order to form a symmetric echo, a padding time is added before 
-and after your tau through a series of delays. 
+and after your tau through a series of delays.
+If you wish to keep the field as is without adjustment, follow
+the 'py run_CPMG.py' command with 'stayput' (e.g. 'py run_CPMG.py stayput')
 """
 from pylab import *
 from pyspecdata import *
+import os, sys
 from numpy import *
 import SpinCore_pp
-from SpinCore_pp.ppg import run_cpmg
-import os
+from SpinCore_pp import prog_plen, get_integer_sampling_intervals
+from SpinCore_pp.ppg import generic
 from datetime import datetime
+from Instruments.XEPR_eth import xepr
 import h5py
-raise RuntimeError("This pulse proram has not been updated.  Before running again, it should be possible to replace a lot of the code below with a call to the function provided by the 'generic' pulse program inside the ppg directory!")
 
-fl = figlist_var()
 # {{{importing acquisition parameters
 config_dict = SpinCore_pp.configuration("active.ini")
-nPoints = int(config_dict["acq_time_ms"] * config_dict["SW_kHz"] + 0.5)
+(
+    nPoints,
+    config_dict["SW_kHz"],
+    config_dict["echo_acq_ms"],
+) = get_integer_sampling_intervals(
+    SW_kHz=config_dict["SW_kHz"], acq_time_ms=config_dict["echo_acq_ms"]
+)
+my_exp_type = "ODNP_NMR_comp/CPMG"
+target_directory = getDATADIR(exp_type=my_exp_type)
+assert os.path.exists(target_directory)
+# }}}
+# {{{ command-line option to leave the field untouched (if you set it once, why set it again)
+adjust_field = True
+if len(sys.argv) == 2 and sys.argv[1] == "stayput":
+    adjust_field = False
+# }}}
+# {{{ let computer set field
+if adjust_field:
+    print(
+        "I'm assuming that you've tuned your probe to",
+        config_dict["carrierFreq_MHz"],
+        "since that's what's in your .ini file",
+    )
```

Comment:
I realize that we want to say this regardless of whether or not we're letting it set the field.

Reply:
In sitting down to run these with warren, I was wondering if it made sense to wait to press enter (so you're not pulsing on unmatched probe) --> that's not objective, but worth thinking about

Reply:
I like this idea and moved outside of conditional block